tests/unittests/BackendTest.cpp

```diff
@@ -188,6 +190,40 @@ TEST_P(BackendTest, CompileWithoutConstants) {
   auto function = backend->compile(F, opts);
 }
 
+/// Test that the runtimeBundle includes only symbols from it's function and not
+/// the whole module.
+TEST_P(BackendTest, BundleFunctionSymbolsOnly) {
+  Module mod;
+  PlaceholderBindings ctx;
+  Function *F = mod.createFunction("main");
+  auto *X = mod.createPlaceholder(ElemKind::FloatTy, {3}, "X", false);
+  auto *XTensor = ctx.allocate(X);
+  XTensor->getHandle() = {1., 2., 3.};
+  auto *pow = F->createPow("Pow1", X, 2.0);
+  auto *save = F->createSave("save", pow);
+  ctx.allocate(save->getPlaceholder());
+  PlaceholderBindings ctx2;
+  Function *F2 = mod.createFunction("main2");
+  auto *X2 = mod.createPlaceholder(ElemKind::FloatTy, {3}, "X2", false);
+  auto *XTensor2 = ctx2.allocate(X2);
+  XTensor2->getHandle() = {1., 2., 3.};
+  auto *pow2 = F2->createPow("Pow2", X2, 2.0);
+  auto *save2 = F2->createSave("save2", pow2);
+  ctx2.allocate(save2->getPlaceholder());
+
+  std::unique_ptr<Backend> backend(createBackend(GetParam()));
+  CompilationOptions opts;
+  auto function = backend->compile(F, opts);
+  auto function2 = backend->compile(F2);
+  auto table1 = function->getRuntimeBundle().getSymbolTable();
+  auto table2 = function2->getRuntimeBundle().getSymbolTable();
+  /// Make sure no symbol in table1 is in table2.
+  for (auto sym : table1) {
+    auto it = table2.find(sym.first);
+    EXPECT_TRUE(it == table2.end());
+  }
```

Comment:
Can you also check that the number of symbols is correct/as expected? Maybe add multiple users to some of the Placeholders/Constants above too to make sure they're not accidentally somehow duplicated in the symbol table?

Reply:
I originally was checking symbol count, but that changes if we run with optimizeIR on or off.
That's why I switch to checking that symbols are not shared when they shouldn't be.